src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -26,8 +32,26 @@ public FindCommand parse(String args) throws ParseException {
         }
 
         String[] nameKeywords = trimmedArgs.split("\\s+");
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_TAG);
+        Set<Tag> tagsFilter = parseTags(argMultimap.getAllValues(PREFIX_TAG));
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)),
+                new ContainsTagsPredicate(tagsFilter));
+    }
+
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<Tag>} if {@code tags} is non-empty.
+     * If {@code tags} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Tag>} containing zero tags.
+     */
+    private Set<Tag> parseTags(Collection<String> tags) throws ParseException {
+        assert tags != null;
+
+        if (tags.isEmpty()) {
+            return Collections.emptySet();
+        }
+        Collection<String> tagSet = tags.size() == 1 && tags.contains("") ? Collections.emptySet() : tags;
+        return ParserUtil.parseTags(tagSet);
```

Comment:
When would tags contain an empty string? Is/should this be a valid input?

Reply:
This part is generally copied from EditCommandParser.java.
Should we throw an error when an empty string is provided?

Reply:
For the original edit command, `t/` with an empty string is an accepted input that clears the tags.
As this is not really a valid input for the `ta/`,`td/` and find commands. I think it would be preferred if we display an error message to the user instead of accepting the input and doing nothing with it.

Reply:
Good call. The error message generated will be the same as inputting an invalid tag, for simplicity and because it is more generalized.

Reply:
Is this fixed? I might have resolved this conversation earlier too quickly